### ISSUE TITLE
Cannot access class '<generated class here>'

### ISSUE THREAD

**Issue by bellini-ea** [google/protobuf-gradle-plugin#471]
Hello,
I'm pretty sure this is just my own local issues but since I'm stumped, thought I would post here incase I'm using the plugin incorrectly or someone has an idea.

I'm hitting this after generating grpc code from the protos and building the project:
`Cannot access class '<my generated class here>'. Check your module classpath for missing or conflicting dependencies`

My setup its a multi module kotlin project, gradle 6.8.2, using the gradle kotlin dsl for the build scripts.  I'm using 0.8.15 version of the protobuf plugin.

**rootproject
   |___ models (subproject)
   |___ services (subproject)**

Both sub projects pull in the same dependency which contains .proto files only.
`protobuf("com.company.project.dcs:dcs-protos:$dcsVersion:proto@zip")`

My gradle code for the protobuf plugin is as follows (this is is both the models project and the services project as they both have the same dependency listed above `protobuf("com.company.....`
```
sourceSets {
    main {
        java {
            srcDirs("$projectDir/src/generated/")
        }
    }
}

protobuf {
    generatedFilesBaseDir = "$projectDir/src/generated"
    tasks { clean { delete(generatedFilesBaseDir) } }

    protoc {
        // The artifact spec for the Protobuf Compiler
        artifact = "com.google.protobuf:protoc:${rootProject.extra.get("protobufVersion") as String}"
    }

    plugins {
        id("grpc-java") {
            artifact = "io.grpc:protoc-gen-grpc-java:${rootProject.extra.get("grpcVersion") as String}"
        }
        id("grpc-kotlin") {
            artifact = "io.grpc:protoc-gen-grpc-kotlin:${rootProject.extra.get("grpcKotlinVersion") as String}"
        }
    }

    generateProtoTasks {
        all().forEach {
            it.plugins {
                id("grpc-java") {
                    outputSubDir = "java"
                }
                id("grpc-kotlin") {
                    outputSubDir = "kotlin"
                }
            }
        }
    }
}
```

**Comment by voidzcy** [comment]
You want want to see if #466 helps you.

**Comment by bellini-ea** [comment]
Thanks for your response.  It gives me clues but the interesting difference is my project does not have either of these dependency
`implementation "com.google.protobuf:protobuf-javalite"`
`implementation "com.google.protobuf:protobuf-java:"`

I tried explicitly adding the `-java` those dependencies as `api` but still get the same problem.  I'm not clear what javalite is and if that has something to do with it.

**Comment by bellini-ea** [comment]
I've created this in a simpler case as the above one was getting unwieldy.  This case is all within the same project, no external dependencies.  Existing code is unable to find the generated code is my issues.  When you include proto{} in your sourceSets, does this both generate the source and add that source to the source set or do you need to explicitly add the generated source to the sourceSets ?  (I tried adding generated source path to the sourceSets but same issue).

**rootproject
|___ protosproject (subproject)
|___ services (subproject)**

The `protosproject` only contains proto files in `src/main/proto` and is a gradle project so they can be published.

`services` project then does the following to get those protos built (all the protoc stuff is in the root project)
```
sourceSets {
    main {
        proto {
            srcDir("${rootProject.projectDir}/protosproject/src/main/proto")
        }
    }
}
```

Result is the following generated code (which is generated) cannot be found
```
Cannot access class 'com.ea.blah.bloop.messages.Contract.DataReply.Builder'. Check your module classpath for missing or conflicting dependencies
```

**Comment by voidzcy** [comment]
> all the protoc stuff is in the root project

I am confused by your project structure. How do you apply the plugin and put the `protobuf{ }` configuration for the `services` module? Can you post the buildscripts for both the root project and `services` module (including the `protobuf{}` block)?

> When you include proto{} in your sourceSets, does this both generate the source and add that source to the source set or do you need to explicitly add the generated source to the sourceSets ? (I tried adding generated source path to the sourceSets but same issue).

When you run the build, the compilation tasks are able to automatically pick up the generated code for proto sources added by the `proto{}` closure. You wouldn't need to explicitly add the path for generated code into the `sourceSets` (although doing that won't break anything).

**Comment by bellini-ea** [comment]
Thanks for all the info.  This issue has been resolved.

I revamped how I was doing this and the issue has been resolved.  I'm not 100% sure, but it was either a duplicate inclusion or something actually not being found din the classpath.  My solution was:
* cleaned up all depedencies between subprojects so they made more sense
* Moved all protobuf {} configuration to the root project
* Removed all sourceSet inclusions from all projects and then added it it only to the projects that complained
* made better use of protobuf dependency type (vs api, implementation), etc
